Subject: Re: [patch 6/6] serial8250: convert to the new platform device interface                                             
From: Olaf Hering <olh@suse.de>                                                                                               

older pSeries systems with serial ports dont get any console output
after recent changes. CONFIG_ISA does not make sense for CONFIG_PPC_PSERIES
because it enables lots of old drivers.
Instead, remove the dependency on CONFIG_ISA from the serial port
discovery code.

Signed-off-by: Olaf Hering <olh@suse.de>

 arch/powerpc/kernel/legacy_serial.c |    4 ----
 1 files changed, 4 deletions(-)

Index: linux-2.6.16-rc1-olh/arch/powerpc/kernel/legacy_serial.c
===================================================================
--- linux-2.6.16-rc1-olh.orig/arch/powerpc/kernel/legacy_serial.c
+++ linux-2.6.16-rc1-olh/arch/powerpc/kernel/legacy_serial.c
@@ -134,7 +134,6 @@ static int __init add_legacy_soc_port(st
 	return add_legacy_port(np, -1, UPIO_MEM, addr, addr, NO_IRQ, flags);
 }
 
-#ifdef CONFIG_ISA
 static int __init add_legacy_isa_port(struct device_node *np,
 				      struct device_node *isa_brg)
 {
@@ -168,7 +167,6 @@ static int __init add_legacy_isa_port(st
 	return add_legacy_port(np, index, UPIO_PORT, reg[1], taddr, NO_IRQ, UPF_BOOT_AUTOCONF);
 
 }
-#endif
 
 #ifdef CONFIG_PCI
 static int __init add_legacy_pci_port(struct device_node *np,
@@ -276,7 +274,6 @@ void __init find_legacy_serial_ports(voi
 		of_node_put(soc);
 	}
 
-#ifdef CONFIG_ISA
 	/* First fill our array with ISA ports */
 	for (np = NULL; (np = of_find_node_by_type(np, "serial"));) {
 		struct device_node *isa = of_get_parent(np);
@@ -287,7 +284,6 @@ void __init find_legacy_serial_ports(voi
 		}
 		of_node_put(isa);
 	}
-#endif
 
 #ifdef CONFIG_PCI
 	/* Next, try to locate PCI ports */
